File: JS/H_One_Prime.js
const fs = require("fs");
const input = fs.readFileSync(0, "utf8").trim().split(/\s+/).map(Number);

// function isPrime(num) {
//     if (num === 1) return false;
//     for (let i = 2; i < num; i++) {
//         if (num % i === 0) return false;
//     }
//     return true;
// }

function isPrime(num) {
    if (num === 1) return false;
    for (let i = 2; i * i <= num; i++) {
        if (num % i === 0) return false;
    }
    return true;
}

const num = input[0];
console.log(isPrime(num) ? "YES" : "NO");

File: C/H_One_Prime.c
#include <stdio.h>
#include <stdbool.h>


// bool isPrime(num)
// {
//     if (num == 1) return false;
//     for (int i = 2; i < num; i++)
//         if (num % i == 0) return false;
//     return true;
// }


bool isPrime(num)
{
    if (num == 1) return false;
    for (int i = 2; i * i <= num; i++)
        if (num % i == 0) return false;
    return true;
}


int main() 
{
    int num;
    scanf("%d", &num);
    if (isPrime(num)) printf("YES\n");
    else printf("NO\n");

    return 0;
}